lib/Controller/HostedSignalingServerController.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Morris Jobke <hey@morrisjobke.de>
+ *
+ * @author Morris Jobke <hey@morrisjobke.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Controller;
+
+use GuzzleHttp\Exception\ClientException;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\IL10N;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\Security\ISecureRandom;
+
+class HostedSignalingServerController extends OCSController {
+
+	/** @var IClientService */
+	protected $clientService;
+	/** @var IL10N */
+	protected $l10n;
+	/** @var IConfig */
+	protected $config;
+	/** @var ISecureRandom */
+	protected $secureRandom;
+	/** @var ILogger */
+	protected $logger;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								IClientService $clientService,
+								IL10N $l10n,
+								IConfig $config,
+								ISecureRandom $secureRandom,
+								ILogger $logger) {
+		parent::__construct($appName, $request);
+		$this->clientService = $clientService;
+		$this->l10n = $l10n;
+		$this->config = $config;
+		$this->secureRandom = $secureRandom;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * @PublicPage
+	 */
+	public function auth(): DataResponse {
+		$storedNonce = $this->config->getAppValue('spreed', 'hosted-signaling-server-nonce', '');
+		// reset nonce after one request
+		$this->config->deleteAppValue('spreed', 'hosted-signaling-server-nonce');
+
+		if ($storedNonce !== '') {
```

Comment:
This should check the sent `nonce` from the request body against what is stored in the database.

Reply:
Why should it? You need to check if the nonce is correct. Otherwise a malicious server would always respond "yes - that is the correct nonce". The idea here is to just send you the last nonce and you do the check.